aiohttp/connector.py

```diff
@@ -394,19 +407,19 @@ def closed(self):
 
             try:
                 await fut
-            except BaseException:
+            except BaseException as e:
                 # remove a waiter even if it was cancelled, normally it's
                 #  removed when it's notified
                 try:
                     waiters.remove(fut)
                 except ValueError:  # fut may no longer be in list
                     pass
 
+                raise e
+            finally:
```

Comment:
@asvetlov can you check this new change? My concern with the previous implementation was because having a `del self._waiters[key]` in two places was a tick bomb and hard to maintain.

Reply:
Looks good